### PR TITLE
Fixs issues in associate a backtrace to erroneous status.(v6d-io#326)

### LINKED ISSUE #326: Associate a backtrace to erroneous Status
**Describe your problem**

Vineyard returns `Status` when error occurs, and we have things like `VINEYARD_CHECK_OK` to act as assertions to check pre/post conditions in both the client and the server.

However, from the `Status` it is still hard to know what unexpected happens and is not every easy for debugging. We have a set of constructors in `Status` to construct a status. It would be good to associate a backtrace into the Status, at least under debug mode.

We have do similar things in [`RETURN_GS_ERROR`](https://github.com/v6d-io/v6d/blob/main/modules/graph/utils/error.h#L121) and we definitely could do more.

**Additional context**

### PR DESCRIPTION
**Pull request by peilii**
What do these changes do?
-------------------------
Associate a backtrace to erroneous Status.
1. Add a `std::string backtrace_` field into the `class Status`
2. Add a getter `const std::string Status::Backtrace() const {}` to `class Status`
3. In constructor of `Status`, when the status code is not “OK", initialize the `backtrace_` field by calling `vineyard::backtrace_info::backtrace()`.

Related issue number
--------------------
Fixes #326

### CODE REVIEW

**Comment by sighingnow** [comment]
Hi @peilii, please sign-off you commit: https://github.com/v6d-io/v6d/pull/335/checks?check_run_id=2880965191

File: src/common/util/status.h [hunk 1]
```diff
@@ -235,6 +235,7 @@ class VINEYARD_MUST_USE_TYPE Status {
     }
   }
   Status(StatusCode, const std::string& msg);
+  Status(StatusCode, const std::string& msg, const std::string& bt);
```
Comment (sighingnow) [inline src/common/util/status.h:238]: That is not need, as we shouldn't expect the user to specify the "backtrace".

**Review by sighingnow — APPROVED** [review]
Just some minor comment, otherwise looks good to me.

**Comment by peilii** [comment]
> Hi @peilii, please sign-off your commit: https://github.com/v6d-io/v6d/pull/335/checks?check_run_id=2880965191

I think it is done now.

File: src/common/util/status.h (hunk 1 above)
Comment (peilii) [reply src/common/util/status.h:238]: > That is not needed, as we shouldn't expect the user to specify the "backtrace".

Sure. This makes sense.

**Comment by sighingnow** [comment]
Merged, thank you @peilii !

**Comment by peilii** [comment]
> Merged, thank you @peilii !

No problem! Thank you for your guidance!

**Pull request merged**